### ISSUE TITLE
doing install got stuck. Could anybody help with this?

### ISSUE THREAD

**Issue by JinranZhang** [lwfinger/rtl8192du#101]
Hi all, I upgrade the kernel of Ubuntu. Then the wifi doesn't work any more. So I hoped to install the wifi adapter again. but after making the source code, I got stuck on the "sudo make install". Could anybody have idea about this? Thanks a lot!
Attached the screen shot.

![20230605_145742](https://github.com/lwfinger/rtl8192du/assets/77756572/0af57987-3479-48a6-8b6c-bcf8cc5d4cd4)

**Comment by lwfinger** [comment]
Do you really need to sign the modules? Doing 'sudo make install' should do the trick.

**Comment by JinranZhang** [comment]
> Do you really need to sign the modules? Doing 'sudo make install' should do the trick.

do you mean to use "sudo make sign-install“？

**Comment by lwfinger** [comment]
Install has only 2 steps - copy the driver to the /lib/modules/ tree  and run depmod on the modules.

What happens when you run 'find /lib/modules/$(uname -r) -name 8723du.ko'?

If you get any output from the command, then the file is there, and it is the depmod that fails.

**Comment by lwfinger** [comment]
I checked agin, and it should log the install command. As it did not, Ubuntu has borked the install command. The workaround would be the following:

sudo mkdir -p /lib/modules/$(uname -r)/kernel/drivers/net/wireless/
sudo cp 8723du.ko /lib/modules/$(uname -r)/kernel/drivers/net/wireless/.
sudo depmod -a $(uname -r)

Runnning those 3 commands should install the driver.

**Comment by JinranZhang** [comment]
> 

Thank you so much! I will try it right now. I reinstalled Ubuntu because I have no idea just now. I need more time to make sure the commands work or not.

**Comment by JinranZhang** [comment]
> I checked agin, and it should log the install command. As it did not, Ubuntu has borked the install command. The workaround would be the following:
> 
> sudo mkdir -p /lib/modules/$(uname -r)/kernel/drivers/net/wireless/ sudo cp 8723du.ko /lib/modules/$(uname -r)/kernel/drivers/net/wireless/. sudo depmod -a $(uname -r)
> 
> Runnning those 3 commands should install the driver.

I upgraded the Ubunu again. when doing install I use the two commands:
sudo cp 8723du.ko /lib/modules/$(uname -r)/kernel/drivers/net/wireless/. 
sudo depmod -a $(uname -r)

then:
sudo modprobe 8192du


uname -r:
5.15.0-73-generic (previous is : 5.4.0-36-generic)

It is better this time. I can see the network sign. but when open "select network" nothing is in it. the light on the wifi adapter was not turned on ( I think it means the adapter doesn't  work)

Do you have any idea?

**Comment by lwfinger** [comment]
On my device, the LED comes on as soon as it is plugged in. Perhaps your device is not in the tables. What does 'lsusb' show for you wireless device?

**Comment by lwfinger** [comment]
Your device is in the tables - it should work.

Go ahead and update, then rebuild and reinstall the driver. An update will likely fix you 'sudo make install' problem.

**Comment by JinranZhang** [comment]
> Go ahead and update, then rebuild and reinstall the driver. An update will likely fix you 'sudo make install' problem.

are you sure? I tried yesterday. updated the software. then kernel became 5.15.0-71-generic from 5.4.0-26-generic. then adapter didnt work. while I tried re-make and re-install the driver for the adapter, it always got stuck. very boring. ..

**Comment by lwfinger** [comment]
Yes, the driver is broken. Whenever you load it, it hangs the system. I am working on trying to get it to work, but I do not know how long it will take. I will let you know. For it to work now, you would need to go back to kernel 5.4.

**Comment by lwfinger** [comment]
I did find a problem. To get it installed, you will need to remove the dongle and reboot. Then do a 'git pull', followed by 'make' and 'sudo make install'. The last one should go to completion. Then plug the dongle back in.

**Comment by JinranZhang** [comment]
> 

wow thank you so much! I will try later. Is it because of the kernel version?

**Comment by lwfinger** [comment]
Yes. The implementation of regd_init() changed.

**Comment by JinranZhang** [comment]
> 

I updated the kernel and reinstall the rtl8192du. When installing, it didn't not get stuck. but after reinstalling the driver, the adapter does not work.

**Comment by lwfinger** [comment]
There is another version for you to try.

While in your rtl8192du directory, do the following:
git checkout origin/v4.0.10 -b v4.0.10
git pull
make
sudo make install

That one should work for you. It does for me.

**Comment by JinranZhang** [comment]
> There is another version for you to try.
> 
> While in your rtl8192du directory, do the following: git checkout origin/v4.0.10 -b v4.0.10 git pull make sudo make install
> 
> That one should work for you. It does for me.

thanks. I did just now. The adapter works now. there is still a little confusion. two adapters are shown (see the attatched photo). they have same name . when I clicked the first one, it doesnt find any available networks. but the second adapter works.
![20230612_151914](https://github.com/lwfinger/rtl8192du/assets/77756572/1bf08798-ae2a-4132-b498-0a4e961c2efa)
